portal/server/routes/middleware/validateMongoId/index.js

```diff
@@ -0,0 +1,20 @@
+const { isValidMongoId } = require('../../../validation/validate-ids');
+
+/**
+ * Redirects the user to '/not-found' if an invalid MongoDB '_id' path param is
+ * provided
+ * @param {object} req - The request object
+ * @param {object} res - The response object
+ * @param {Function} next - The next middleware function
```

Comment:
I've started using different types for these as of late just because we know this is using express:
```
* @ param {import('express').Request} req - Express request object
* @ param {import('express').Response} res - Express response object
* @ param {import('express').NextFunction} req - Express next function
```

Reply:
There are a few more places where the JSDocs have been written like this in this PR, so if you agree with me then don't forget to change it in those places too :)

Reply:
Agreed, this is much better. We'll need to add `@types/express` to the project though for the IDE to actually recognise these types and give us some type safety and auto-complete. Will add as part of this PR 👍